another attempt to fix launch command support for Windows

seanyen opened the pull request:
It turned out the shlex.split with posix=False won't deal with double quotes as what posix mode does. And shlex.split with posix=True won't handle the backslash correctly, so both mode doesn't output any worked string for Windows to run. And in fact, Popen on Windows can just take the whole command as a string to run. So here is an attempt to remove the usage of shlex.split on Windows.

seanyen committed:
another attempt to fix launch command support for Windows

ooeygui commented:
How does this manifest? (in other words, how did you find it and verify the fix?)

johnsonshih commented on tools/roslaunch/src/roslaunch/loader.py:
```diff
@@ -500,8 +500,9 @@ def param_value(self, verbose, name, ptype, value, textfile, binfile, command):
             try:
-                os_posix = os.name == "posix"
-                p = subprocess.Popen(shlex.split(command, posix=os_posix), stdout=subprocess.PIPE)
+                if os.name == "posix":
```
>os.name == "posix": [](start = 19, length = 19)

The same check posfix=os_posfox was added to multiple locations, do we need to revise those changes, too?

seanyen commented:
@ooeygui The original problem we run into is that the shlex.split (with posix=True) will ignore backslash and convert a string like this "c:\abc\def\ghi" into "c:abcdefghi" and the resulting string is just wrong to run. And shortly after my posix=False fix for Windows, @johnsonshih noticed there is a roslaunch unit test failure when the input command comes with double quotes in the parameters, for example, `abc "def"` and then shlex.split will add unnecessary escape characters to the double quotes and it ends up a string like `\"def\"`. For better coverage, we can add more Windows-style command lines test cases, so far I am testing it against the exsting unit tests and some example launch files from other packages (for example, moveit).

seanyen committed:
move the winsock2.h into cpp.

seanyen committed:
Revert "move the winsock2.h into cpp."

This reverts commit de4268ad38ef0052d98c22417a0ca1f1bd38c4a3.
